### PR TITLE
chore!: remove node 10 support

### LINKED ISSUE #3512: Upgrade minimal Node.js version to Node 12
See background at https://github.com/netlify/team-dev/issues/34

We should upgrade the minimal Node.js version to Node `>=12.20.0` and make a major release.

Since we can make a major release, upgrading would require a user action and no sites would be broken.

### PR DESCRIPTION
**Pull request by lukasholzer**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Fixes #3512

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [x] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

image

### CODE REVIEW

File: src/functions-templates/javascript/google-analytics/package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
   "author": "Netlify",
   "license": "MIT",
   "dependencies": {
+    "netlify-cli": "latest",
```
Comment (erezrokah) [inline src/functions-templates/javascript/google-analytics/package.json:22]: What is the reason we need the CLI as a dependency for this function template?
Function templates are used for this command:
https://cli.netlify.com/commands/functions#functionscreate

to create a new function from a template.

File: tests/command.deploy.test.js [hunk 2]
```diff
@@ -92,6 +92,7 @@ if (process.env.NETLIFY_TEST_DISABLE_LIVE !== 'true') {
     })
   })
 
+  // TODO: check with erez if we should update edge handler min version to 12 so they can drop compatibility
```
Comment (erezrokah) [inline tests/command.deploy.test.js:95]: For the CLI tests we should drop this condition (we should have dropped it a while back).

I suggest opening a new issue in https://github.com/netlify/netlify-plugin-edge-handlers to handle Node.js 10 deprecation

File: tests/command.dev.test.js [hunk 3]
```diff
@@ -1310,6 +1310,7 @@ testMatrix.forEach(({ args }) => {
     })
   })
 
+  // TODO: check with erez if we should update edge handler min version to 12 so they can drop compatibility
```
Comment (erezrokah) [inline tests/command.dev.test.js:1313]: Same as previous comment

**Review by erezrokah — CHANGES_REQUESTED** [review]
This is looking 🔥 @lukasholzer, answered your question about Edge Handlers and added a comment.

**Comment by ehmicky** [comment]
This is looking great to me too! :+1: 
Nothing to add except for what @erezrokah mentioned, with one exception: the CI tests are currently failing, which might indicate some problem.

**Comment by lukasholzer** [comment]
> This is looking great to me too! 👍 Nothing to add except for what @erezrokah mentioned, with one exception: the CI tests are currently failing, which might indicate some problem.

Yea currently figuring out why the tests are failing – therefore I set the PR in the draft state. This should indicate that it is not ready for review – I try to create always fast PR's so that folks that are interested can already see in which direction it turns, fast feedback. 

But it is strange that it is failing as my changes should not influence that

**Comment by lukasholzer** [comment]
The `build (ubuntu-latest, 10.x)` is hanging around as there is now a new `Testing for CLI / build (ubuntu-latest, 12.x)` that we need to add as required status checks once this PR is merged

**Review by erezrokah — APPROVED** [review]

**Comment by erezrokah** [comment]
> The `build (ubuntu-latest, 10.x)` is hanging around as there is now a new `Testing for CLI / build (ubuntu-latest, 12.x)` that we need to add as required status checks once this PR is merged

I updated the branch protections settings, so this should be fixed now

**Comment by erezrokah** [comment]
> I updated the branch protections settings, so this should be fixed now

You should be able to do it too I think as you're a part of https://github.com/orgs/netlify/teams/netlify-dev

**Review by ehmicky — APPROVED** [review]
:tada:

**Pull request merged**